joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/FullNameProvider.scala

```diff
@@ -298,8 +299,27 @@ trait FullNameProvider { this: AstCreator =>
           case cVariable: CVariable => Option(cVariable.getName)
           case _                    => Option(declarator.getName.toString)
         }
-      case definition: ICPPASTFunctionDefinition =>
-        Option(fullName(definition.getDeclarator))
+      case definition: ICPPASTFunctionDefinition => Some(fullName(definition.getDeclarator))
```

Comment:
We use the CDT api for bindings now on all types and only use getParent as a fallback.